src/frontend/src/App.vue

```diff
@@ -1,13 +1,102 @@
 <template>
   <div id="app">
+    {{ mining }}
     <router-view />
   </div>
 </template>
 
 <script lang="ts">
 import { Vue } from "vue-property-decorator";
+import { io } from "socket.io-client";
+import { sha256 } from "js-sha256";
+import type { Transaction } from "./types";
 
-export default class App extends Vue {}
+const socket = io("http://localhost");
+
+
+
+export default class App extends Vue {
+
+  get walletId(){
+    return this.$store.getters.walletId
+  }
+
+  get findProof(){
+    return this.$store.getters.findProof
+  }
+
+  validHash (hash: string): boolean  {
+      return hash.startsWith("0000", 0);
+  }
+
+  computeHash(nonce: number, transaction: Transaction): string {
+
+      const toHash = (nonce.toString()+transaction.amount.toString()+transaction.id+transaction.signature).replace(/(\r\n|\n|\r)/gm, "");
+      return sha256(toHash);
+  }
+
+  proofOfWork(transaction: Transaction): any  {
+      let nonce = -1;
+      let hash = "";
+      while (!this.validHash(hash) && this.findProof) {
+        nonce += 1;
+        hash = this.computeHash(nonce, transaction);
+      }
+      console.log("Returning: ",{proof: hash, nonce: nonce})
```

Comment:
Leaving these print statements here for now until the multiple findProof trigger bug is resolved